src/static_array.cr

```diff
@@ -86,6 +86,21 @@ struct StaticArray(T, N)
     to_unsafe.shuffle!(size)
     self
   end
+  
+
+  # Returns a new array with elements of self shuffled
+  # The optional rng argument will be used as the random number generator.
+  #
+  # ```
+  # a = StaticArray(Int32, 3).new { |i| i + 1 } # => [1, 2, 3]
+  # a.shuffle!(Random.new(42))                  # => [3, 2, 1]
+  # a                                           # => [3, 2, 1]
+  # ```
+  def shuffle!(rng : Random)
```

Comment:
I would modify the existing `shuffle!` method instead of adding an overload, so exactly the same as in `Array#shuffle!`. Also, it modifies the array in place, so "returns a new array" is not quite right.